src/Components/ProductListing.js

```diff
@@ -0,0 +1,204 @@
+              <fieldset>
+                <legend className="card-title">Filter</legend>
+                <p className="card-text">
+                  <input
+                    name="outofstock"
+                    type="checkbox"
+                    checked={showInventoryAll}
+                    onChange={() => dispatch({ type: "TOGGLE_INVENTORY" })}
+                  />
+                  <label htmlFor="outofstock">Include Out of Stock</label>
+                  <br />
+                  <input
+                    checked={showFastDeliveryOnly}
+                    name="fastdelivery"
+                    type="checkbox"
+                    onChange={() => dispatch({ type: "TOGGLE_DELIVERY" })}
+                  />
+                  <label htmlFor="fastdelivery">Fast Delivery</label>
+                </p>
+              </fieldset>
+            </div>
+          </div>
+        </div>
+      </div>
+      <div className="something">
+        <h1>Our Products</h1>
+        <ul className="cards">
+          {filteredData.map((item) => {
+            // console.log(item.count);
+            let btn_class = item.wishlist ? "icon-btn-red" : "icon-btn";
+            let btn_type = item.wishlist ? "favorite" : "favorite_border";
+            return (
+              <li className="cards-item" key={item.id}>
+                <div className="card">
+                  <>
+                    <img src={item.image} alt="randomimage" />
+                    <button
+                      className={btn_class}
+                      onClick={() => {
+                        let temp = items.map((el) =>
+                          el.id === item.id
+                            ? { ...el, wishlist: !el.wishlist }
+                            : el
+                        );
+                        console.log();
+                        setItems(temp);
+                      }}
+                    >
+                      <i className="material-icons-outlined md-36">
+                        {btn_type}
+                      </i>
+                    </button>
+                  </>
+                  <div className="card-content">
+                    <h2 className="card-title">{item.name}</h2>
+                    <div className="card-text">
+                      Price in Rs: <strong>{item.price}</strong>
+                    </div>
+                    {item.inStock ? (
+                      <div>In Stock</div>
+                    ) : (
+                      <div>Out Of Stock</div>
+                    )}
+                    {item.fastDelivery ? (
+                      <div>Fast Delivery</div>
+                    ) : (
+                      <div>3 Days Minimum</div>
+                    )}
+
+                    {!item.count ? (
+                      <button
+                        disabled={!item.inStock}
+                        style={
+                          !item.inStock
+                            ? {
```

Comment:
Have a disabled button class in your component library ,and conditionally check the button classname